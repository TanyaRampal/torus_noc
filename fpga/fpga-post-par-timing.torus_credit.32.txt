Copyright 1986-2022 Xilinx, Inc. All Rights Reserved.
-----------------------------------------------------------------------------------------
| Tool Version : Vivado v.2022.1 (lin64) Build 3526262 Mon Apr 18 15:47:01 MDT 2022
| Date         : Fri Dec  6 14:42:17 2024
| Host         : ECEUBUNTU2 running 64-bit Red Hat Enterprise Linux release 8.10 (Ootpa)
| Command      : report_timing -file fpga-post-par-timing.torus_credit.32.txt
| Design       : torus_credit
| Device       : 7z020-clg400
| Speed File   : -1  PRODUCTION 1.12 2019-11-22
-----------------------------------------------------------------------------------------

Timing Report

Slack (VIOLATED) :        -1.554ns  (required time - arrival time)
  Source:                 ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[1].vc_fifo/fifo_tail_reg[2]/C
                            (rising edge-triggered cell FDRE clocked by clk  {rise@0.000ns fall@3.000ns period=6.000ns})
  Destination:            ys[0].xs[3].torus_switch_xy/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_24_29/RAMA/WE
                            (rising edge-triggered cell RAMD32 clocked by clk  {rise@0.000ns fall@3.000ns period=6.000ns})
  Path Group:             clk
  Path Type:              Setup (Max at Slow Process Corner)
  Requirement:            6.000ns  (clk rise@6.000ns - clk rise@0.000ns)
  Data Path Delay:        6.937ns  (logic 1.436ns (20.701%)  route 5.501ns (79.299%))
  Logic Levels:           6  (LUT4=1 LUT6=4 RAMD32=1)
  Clock Path Skew:        -0.049ns (DCD - SCD + CPR)
    Destination Clock Delay (DCD):    0.924ns = ( 6.924 - 6.000 ) 
    Source Clock Delay      (SCD):    0.973ns
    Clock Pessimism Removal (CPR):    0.000ns
  Clock Uncertainty:      0.035ns  ((TSJ^2 + TIJ^2)^1/2 + DJ) / 2 + PE
    Total System Jitter     (TSJ):    0.071ns
    Total Input Jitter      (TIJ):    0.000ns
    Discrete Jitter          (DJ):    0.000ns
    Phase Error              (PE):    0.000ns

    Location             Delay type                Incr(ns)  Path(ns)    PBlock               Netlist Resource(s)
  -------------------------------------------------------------------    ----------------------------------------
                         (clock clk rise edge)        0.000     0.000 r                       
                                                      0.000     0.000 r                       clk (IN)
                         net (fo=5113, unset)         0.973     0.973                         ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[1].vc_fifo/clk
    SLICE_X105Y26        FDRE                                         r  poly2_0_sw           ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[1].vc_fifo/fifo_tail_reg[2]/C
  -------------------------------------------------------------------    ----------------------------------------
    SLICE_X105Y26        FDRE (Prop_fdre_C_Q)         0.456     1.429 r  poly2_0_sw           ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[1].vc_fifo/fifo_tail_reg[2]/Q
                         net (fo=43, routed)          0.988     2.417                         ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[1].vc_fifo/fifo_data_reg_0_31_30_35/ADDRC2
    SLICE_X104Y26        RAMD32 (Prop_ramd32_RADR2_O)
                                                      0.153     2.570 f  poly2_0_sw           ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[1].vc_fifo/fifo_data_reg_0_31_30_35/RAMC/O
                         net (fo=1, routed)           0.980     3.549                         ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[2].vc_fifo/o_rdata[34]
    SLICE_X102Y27        LUT6 (Prop_lut6_I2_O)        0.331     3.880 f  poly2_0_sw           ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[2].vc_fifo/s_out_x_reg[0]_i_2__0/O
                         net (fo=5, routed)           0.507     4.387                         ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[2].vc_fifo/w_in_x[0]
    SLICE_X103Y27        LUT4 (Prop_lut4_I0_O)        0.124     4.511 r  poly2_0_sw           ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[2].vc_fifo/grant_base[2]_i_10__0/O
                         net (fo=9, routed)           0.490     5.001                         ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[2].vc_fifo/dor_credit_inst/e_v117_out
    SLICE_X103Y28        LUT6 (Prop_lut6_I4_O)        0.124     5.125 r  poly2_0_sw           ys[0].xs[2].torus_switch_xy/west_conn_rx/gen_vc_logic[2].vc_fifo/waiting_for_ack_i_4__1/O
                         net (fo=4, routed)           0.644     5.770                         ys[0].xs[2].client_xy/e_v114_out
    SLICE_X102Y26        LUT6 (Prop_lut6_I3_O)        0.124     5.894 r  poly2_0_cli          ys[0].xs[2].client_xy/fifo_data_reg_0_31_0_5_i_8__0/O
                         net (fo=2, routed)           0.174     6.067                         ys[0].xs[2].torus_switch_xy/east_conn_tx/e_v[0]
    SLICE_X102Y26        LUT6 (Prop_lut6_I0_O)        0.124     6.191 r  poly2_0_sw           ys[0].xs[2].torus_switch_xy/east_conn_tx/fifo_data_reg_0_31_0_5_i_1__5/O
                         net (fo=59, routed)          1.718     7.910                         ys[0].xs[3].torus_switch_xy/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_24_29/WE
    SLICE_X66Y29         RAMD32                                       r  poly3_0_sw           ys[0].xs[3].torus_switch_xy/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_24_29/RAMA/WE
  -------------------------------------------------------------------    ----------------------------------------

                         (clock clk rise edge)        6.000     6.000 r                       
                                                      0.000     6.000 r                       clk (IN)
                         net (fo=5113, unset)         0.924     6.924                         ys[0].xs[3].torus_switch_xy/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_24_29/WCLK
    SLICE_X66Y29         RAMD32                                       r  poly3_0_sw           ys[0].xs[3].torus_switch_xy/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_24_29/RAMA/CLK
                         clock pessimism              0.000     6.924                           
                         clock uncertainty           -0.035     6.889                           
    SLICE_X66Y29         RAMD32 (Setup_ramd32_CLK_WE)
                                                     -0.533     6.356    poly3_0_sw             ys[0].xs[3].torus_switch_xy/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_24_29/RAMA
  -------------------------------------------------------------------
                         required time                          6.356                           
                         arrival time                          -7.910                           
  -------------------------------------------------------------------
                         slack                                 -1.554                           




