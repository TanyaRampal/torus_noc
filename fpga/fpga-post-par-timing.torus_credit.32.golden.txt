Copyright 1986-2022 Xilinx, Inc. All Rights Reserved.
-----------------------------------------------------------------------------------------
| Tool Version : Vivado v.2022.1 (lin64) Build 3526262 Mon Apr 18 15:47:01 MDT 2022
| Date         : Tue Nov 26 14:39:58 2024
| Host         : ECEUBUNTU2 running 64-bit Red Hat Enterprise Linux release 8.10 (Ootpa)
| Command      : report_timing -file fpga-post-par-timing.torus_credit.32.txt
| Design       : torus_credit
| Device       : 7z020-clg400
| Speed File   : -1  PRODUCTION 1.12 2019-11-22
-----------------------------------------------------------------------------------------

Timing Report

Slack (MET) :             0.615ns  (required time - arrival time)
  Source:                 ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[2].vc_fifo/fifo_tail_reg[1]/C
                            (rising edge-triggered cell FDRE clocked by clk  {rise@0.000ns fall@4.500ns period=9.000ns})
  Destination:            ys[1].xs[3].switch/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_0_5/RAMC/I
                            (rising edge-triggered cell RAMD32 clocked by clk  {rise@0.000ns fall@4.500ns period=9.000ns})
  Path Group:             clk
  Path Type:              Setup (Max at Slow Process Corner)
  Requirement:            9.000ns  (clk rise@9.000ns - clk rise@0.000ns)
  Data Path Delay:        7.923ns  (logic 1.676ns (21.154%)  route 6.247ns (78.846%))
  Logic Levels:           5  (LUT2=1 LUT3=1 LUT6=2 RAMD32=1)
  Clock Path Skew:        -0.049ns (DCD - SCD + CPR)
    Destination Clock Delay (DCD):    0.924ns = ( 9.924 - 9.000 ) 
    Source Clock Delay      (SCD):    0.973ns
    Clock Pessimism Removal (CPR):    0.000ns
  Clock Uncertainty:      0.035ns  ((TSJ^2 + TIJ^2)^1/2 + DJ) / 2 + PE
    Total System Jitter     (TSJ):    0.071ns
    Total Input Jitter      (TIJ):    0.000ns
    Discrete Jitter          (DJ):    0.000ns
    Phase Error              (PE):    0.000ns

    Location             Delay type                Incr(ns)  Path(ns)    PBlock               Netlist Resource(s)
  -------------------------------------------------------------------    ----------------------------------------
                         (clock clk rise edge)        0.000     0.000 r                       
                                                      0.000     0.000 r                       clk (IN)
                         net (fo=5647, unset)         0.973     0.973                         ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[2].vc_fifo/clk
    SLICE_X103Y88        FDRE                                         r  poly2_1_sw           ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[2].vc_fifo/fifo_tail_reg[1]/C
  -------------------------------------------------------------------    ----------------------------------------
    SLICE_X103Y88        FDRE (Prop_fdre_C_Q)         0.419     1.392 r  poly2_1_sw           ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[2].vc_fifo/fifo_tail_reg[1]/Q
                         net (fo=46, routed)          1.138     2.530                         ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[2].vc_fifo/fifo_data_reg_0_31_30_35/ADDRC1
    SLICE_X100Y91        RAMD32 (Prop_ramd32_RADR1_O)
                                                      0.328     2.858 r  poly2_1_sw           ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[2].vc_fifo/fifo_data_reg_0_31_30_35/RAMC/O
                         net (fo=1, routed)           0.745     3.603                         ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[1].vc_fifo/o_v_r_i_3__5_0[0]
    SLICE_X99Y87         LUT6 (Prop_lut6_I4_O)        0.331     3.934 f  poly2_1_sw           ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[1].vc_fifo/ys[1].xs[2].conns_addr[2].e_tx_addr_i_2/O
                         net (fo=4, routed)           0.628     4.563                         ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[1].vc_fifo/genrr.grant_reg[1]_0
    SLICE_X101Y88        LUT2 (Prop_lut2_I0_O)        0.118     4.681 f  poly2_1_sw           ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[1].vc_fifo/ys[1].xs[2].conns_vc_info[0].e_tx_vc_i_4/O
                         net (fo=9, routed)           0.695     5.376                         ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_tail_reg[4]_0
    SLICE_X105Y88        LUT6 (Prop_lut6_I4_O)        0.326     5.702 r  poly2_1_sw           ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[0].vc_fifo/ys[1].xs[2].conns_data[0].e_tx_data_i_3/O
                         net (fo=36, routed)          1.100     6.802                         ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[2].vc_fifo/fifo_data_reg_0_31_30_35_0
    SLICE_X106Y90        LUT3 (Prop_lut3_I1_O)        0.154     6.956 r  poly2_1_sw           ys[1].xs[2].switch/west_conn_rx/gen_vc_logic[2].vc_fifo/ys[1].xs[2].conns_data[4].e_tx_data_i_1/O
                         net (fo=3, routed)           1.940     8.896                         ys[1].xs[3].switch/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_0_5/DIC0
    SLICE_X50Y88         RAMD32                                       r  poly3_1_sw           ys[1].xs[3].switch/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_0_5/RAMC/I
  -------------------------------------------------------------------    ----------------------------------------

                         (clock clk rise edge)        9.000     9.000 r                       
                                                      0.000     9.000 r                       clk (IN)
                         net (fo=5647, unset)         0.924     9.924                         ys[1].xs[3].switch/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_0_5/WCLK
    SLICE_X50Y88         RAMD32                                       r  poly3_1_sw           ys[1].xs[3].switch/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_0_5/RAMC/CLK
                         clock pessimism              0.000     9.924                           
                         clock uncertainty           -0.035     9.889                           
    SLICE_X50Y88         RAMD32 (Setup_ramd32_CLK_I)
                                                     -0.378     9.511    poly3_1_sw             ys[1].xs[3].switch/west_conn_rx/gen_vc_logic[0].vc_fifo/fifo_data_reg_0_31_0_5/RAMC
  -------------------------------------------------------------------
                         required time                          9.511                           
                         arrival time                          -8.896                           
  -------------------------------------------------------------------
                         slack                                  0.615                           




